### DISCUSSION TITLE
How to do Master-Public key rotation?

### DISCUSSION THREAD

**Discussion by HvShirishBhalekar** [openssl/openssl#25872] [Q&A]
Is there is any existing function like seal functions can do below things in openssl inside one function only.

I know below procedure 

1. Generate New Keys 
2. Decrypt the DEK using old one of private key
3. Decrypt the data using DEK.
4. Encrypt the data using new DEK and IV.
5. Encrypt the DEK using new Public Keys.

**Comment by tom-cosgrove-arm** [comment]
This question should really be asked as a discussion - it is not an issue (aka problem) with OpenSSL

**Reply by HvShirishBhalekar** [reply]
Anything helpful can be good
